virttest/libvirtd_decorator.py

```diff
@@ -0,0 +1,70 @@
+"""
+A decorator utility functions to apply libvirtd functions.
+
+:copyright: 2008-2009 Red Hat Inc.
+"""
+
+import logging
+import os
+from virttest import data_dir
+from virttest import libvirt_version
+
+
+def get_libvirtd_split_enable_bit():
+    base_cfg_path = os.path.join(data_dir.get_shared_dir(), 'cfg', 'base.cfg')
+    if os.path.isfile(base_cfg_path):
+        with open(base_cfg_path, 'r') as base_file:
+            for line in base_file:
+                if 'enable_split_libvirtd_feature' in line and 'yes' in line:
+                    return True
+    else:
+        logging.info("CAN NOT find base.cfg file")
+    return False
+
+
+LIBVIRD_SPLIT_ENABLE_BIT = get_libvirtd_split_enable_bit()
+IS_LIBVIRTD_SPLIT_VERSION = libvirt_version.version_compare(5, 6, 0)
+
+
+def libvirt_version_context_aware_libvirtd_legacy(fn):
+    """
+    A decorator that must be applied to functions that call if libvirt version <5.6.0.
+
+    :param fn: function name.
+    """
+    def new_fn(*args, **kwargs):
+        """
+        Keep previous function as working before if libvirt version< 5.6.0, else do nothing
+
+        :param args: function fixed args.
+        :param kwargs: function varied args.
+        """
+        if not IS_LIBVIRTD_SPLIT_VERSION or not LIBVIRD_SPLIT_ENABLE_BIT:
+            logging.warning("legacy start libvirtd daemon NORMALLY with function name: %s" % fn.__name__)
+            return fn(*args, **kwargs)
+        else:
+            logging.warning("legacy start libvirtd daemon IGNORED with function name: %s" % fn.__name__)
+            return None
```

Comment:
I assume this None will be replaced by all split dameon's respective function call?

Reply:
current solution is to keep return None since we can not determine which split dameon(virtqemud,virtproxyd,virtneworkd,etc) is in the acting at runtime.
And the decorator will apply legacy libvirtd operations such as start,stop,restart  in order to ignore those effects if module daemons mode is running.